src/Components/CookieConsentContext.js

```diff
@@ -111,9 +99,11 @@ export function CookieConsentProvider({
   const isCookieConsentEmpty = () =>
     Object.keys(cookieConsentChoice).length === 0;
 
-  const saveAndClose = () => {
-    if (isCookieConsentEmpty()) {
-      setDecisionForAllCookies(DECLINED);
+  const saveAndClose = (decision) => {
+    if (decision === ACCEPTED || decision === DECLINED) {
+      setCookieDecision(editableCookies, decision);
+    } else if (isCookieConsentEmpty()) {
+      setCookieDecision(editableCookies, DECLINED);
```

Comment:
was this an issue?

Reply:
No, this was just a little refactoring. I just thought that 3 function that do the same are not necessary.